x/log_writer.go

```diff
@@ -404,7 +406,8 @@ func processOldLogFiles(fp string, maxAge int64) ([]string, []string, error) {
 		}
 
 		_, e := prefixAndExt(fp)
-		ts, err := time.Parse(backupTimeFormat, f.Name()[len(defPrefix):len(f.Name())-len(e)])
+		tsString := f.Name()[len(defPrefix) : len(f.Name())-len(e)]
+		ts, err := time.ParseInLocation(backupTimeFormat, tsString, time.Local)
```

Comment:
@ajeetdsouza can you please point me to the code that creates the file? I can't find that code.

Reply:
https://github.com/dgraph-io/dgraph/blob/4e4842624d26bb24caa08019d7be834bc59e282b/x/log_writer.go#L302-L307

I don't think we're writing the time in Local format.

Reply:
@jarifibrahim `time.Now()` returns time using the local timezone, but `backupTimeFormat` does not include timezone information, which is what is causing the problem.

Sample code:

```go
package main

import (
	"fmt"
	"time"
)

func main() {
	const backupTimeFormat = "2006-01-02T15-04-05.000"
	now := time.Now()
	tsLocal := now.Format(backupTimeFormat)
	tsUTC := now.UTC().Format(backupTimeFormat)
	fmt.Printf("tsLocal: %s\ntsUTC: %s\n", tsLocal, tsUTC) // different values
}
```

There are three alternate approaches here:
- Change `backupTimeFormat` to include timezone.
- Use UTC when formatting to `backupTimeFormat`.
- Both of the above.

I think these would be better than the current approach, since there is no ambiguity that way.

Reply:
>Change backupTimeFormat to include timezone.
Use UTC when formatting to backupTimeFormat.
Both of the above.

This would change the file name and we might not be able to parse it again. Is this backupTimeFormat used only in audit logs? @aman-bansal why is it called backupTimeFormat?

If this is used only in audit logs, we can use time.UTC() everywhere for simplicity.

Reply:
@jarifibrahim `backupTimeFormat` is only used within `LogWriter`, so I think it's safe to switch over to UTC timestamps. I just changed it to use UTC everywhere.